aleo-programs/staking_lite/src/main.leo

```diff
@@ -221,7 +312,6 @@ program staking_lite.aleo {
   ) -> credits.aleo/credits {
     // Must be a credits record because credits.aleo uses self.caller for transfers
     let updated_record: credits.aleo/credits = credits.aleo/transfer_private_to_public(input_record, CORE_PROTOCOL, microcredits);
-    credits.aleo/bond_public(validator_address, microcredits);
```

Comment:
I got rid of bond_public from the deposit. The thinking here is that the validator or anyone else can always call: `bond_all`. This will effectively batch deposits and we can get rid of the state checks which would have prevent from depositing while people were withdrawing (which I don't think would fly with validators).